src/gui/properties/speedplotview.h

```diff
@@ -0,0 +1,51 @@
+#ifndef SPEEDPLOTVIEW_H
+#define SPEEDPLOTVIEW_H
+
+#include <QGraphicsView>
+#include <QPen>
+#include <QQueue>
+#include <QMap>
+
+class SpeedPlotView : public QGraphicsView
+{
+    Q_OBJECT
+public:
+    explicit SpeedPlotView(QWidget *parent = 0);
+
+    void setGraphName(int id, const QString &name);
+    void setGraphPen(int id, const QPen &pen);
+    void setGraphEnable(int id, bool enable);
+
+    void pushXPoint(double x);
+    void pushYPoint(int id, double y);
+
+    void setViewableLastPoints(int period);
+    void setMaxCapacity(int capacity);
+
+    void replot();
+
+protected:
+    virtual void paintEvent(QPaintEvent *event);
+
+private:
+    struct GraphProperties
```

Comment:
Move this struct declaration into the .cpp file.
And just do here `struct GraphProperties`. I hope this forward declaration works.